ffi/src/lib.rs

```diff
@@ -568,14 +568,31 @@ pub unsafe extern "C" fn snapshot(
 ) -> ExternResult<Handle<SharedSnapshot>> {
     let url = unsafe { unwrap_and_parse_path_as_url(path) };
     let engine = unsafe { engine.as_ref() };
-    snapshot_impl(url, engine).into_extern_result(&engine)
+    snapshot_impl(url, engine, None).into_extern_result(&engine)
+}
+
+/// Get the snapshot from the specified table at a specific version
+///
+/// # Safety
+///
+/// Caller is responsible for passing valid handles and path pointer.
+#[no_mangle]
+pub unsafe extern "C" fn snapshot_at_version(
+    path: KernelStringSlice,
+    engine: Handle<SharedExternEngine>,
+    version: u64,
```

Comment:
do things work if we make this `version: Version` (i.e. is cbindgen clever enough to translate that?)

I'd prefer to have this change too if we ever change the type for `Version`